proxy/interceptor.go

```diff
@@ -97,8 +98,22 @@ func NewPayloadVisitorInterceptor(options PayloadVisitorInterceptorOptions) (grp
 		}
 
 		err := invoker(ctx, method, req, response, cc, opts...)
-		if err != nil {
-			return err
+		if err != nil && options.Inbound != nil {
+			stat, ok := status.FromError(err)
+			if ok {
+				// user provided payloads can sometimes end up in the status details of
+				// gRPC errors, make sure to visit those as well
+				for _, detail := range stat.Details() {
+					detailAny, ok := detail.(*anypb.Any)
+					if !ok {
+						return fmt.Errorf("Error returned from rpc invoker should be anypb.Any")
+					}
+					if detailAny.MessageName() == "temporal.api.errordetails.v1.QueryFailedFailure" || detailAny.MessageName() == "temporal.api.errordetails.v1.MultiOperationExecutionFailure" {
+						VisitPayloads(ctx, detailAny, *options.Inbound)
+					}
```

Comment:
```suggestion
					if ok && (detailAny.MessageName() == "temporal.api.errordetails.v1.QueryFailedFailure" || detailAny.MessageName() == "temporal.api.errordetails.v1.MultiOperationExecutionFailure") {
						VisitPayloads(ctx, detailAny, *options.Inbound)
					}
```

Do not return our own custom errors here if the types do not conform